### DISCUSSION TITLE
How to set lookup value in typescript using a generic method

### DISCUSSION THREAD

**Discussion by wchitirobho** [serenity-is/Serenity#6655] [Help] [answered]
Good day guys.

I am trying to create a generic client side caching method in case a user of my system does not hit save button or the page timeout before they save, I would want to restore from browser cache once they try to create a new record again

I could obviously use this syntax:  **this.form.LOOKUPFIELD.set_value(t1id);** but I am looking for a more generic method of how to set a lookup value.

my code below:

        protected afterLoadEntity() {
            super.afterLoadEntity();

            let thisForm = this;

            $(() => {
                thisForm.BindCache()
            });

        }

        protected BindCache() {
            let thisForm = this;
            let LRNSerial = this.form.LrnSerial.value;
            if (Q.isEmptyOrNull(LRNSerial)) {
                LRNSerial = "";
            }

            $(".s-LookupEditor").each(function (index) {
                let splitvalues = this.id.split('_');
                let sfieldname = splitvalues[splitvalues.length - 1];

                    let val = thisForm.cache.getItem(LRNSerial + sfieldname);
                    if (Q.isEmptyOrNull($(this).val()) || $(this).val().toLowerCase() == "null") {
                        $(this).val(val);

                    }
                    console.log(LRNSerial + sfieldname + ' > ' + val);
            });

            $('.s-LookupEditor').on('change', function () {
                let splitvalues = this.id.split('_');
                let sfieldname = splitvalues[splitvalues.length - 1];

                thisForm.cache.setItem(LRNSerial + sfieldname, $(this).val());

                let val = thisForm.cache.getItem(LRNSerial + sfieldname);
                console.log(LRNSerial + sfieldname + ' > ' + val);

            });
        }

Thanks in advance.

![Cache Update](https://user-images.githubusercontent.com/5858493/212263259-ff4ef3c3-4fe3-4503-b579-f792bfd55427.png)

**Comment by wezmag** [comment, marked as answer]
The lookup editor is using jQuery Select2 plugin.
Maybe you can do it with [Select2 methods](https://select2.github.io/select2/#programmatic)

**Comment by wchitirobho** [comment]
@wezmag Thanks very much it worked very well, so now if I capture data on my form and I do not save and I come back to the same for to create a new record, my previous values populate. GREAT !!!

![image](https://user-images.githubusercontent.com/5858493/212296246-707bb313-5906-4a73-95f9-4a025d70c956.png)
